mautic/mautic
# Update grapesjs to more recent version (Mautic 5)

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

_**[The equivalent for M4 is here](https://github.com/mautic/mautic/pull/13285)**_

This updates GrapesJs to the latest version (at time of writing) as well as its dependencies.
It also updates Parcel (the build tool) to v.2 (v1 is no longer supported)
Because it relies on an update of `grapesjs-preset-mautic`, and that hasn't been merged yet ([see this PR](https://github.com/mautic/grapesjs-preset-mautic/pull/49)), I loaded [my forked branch](https://github.com/mautic/mautic/pull/13283/files#diff-7bfbc6d869d142cddd9b934e5de8527cbe7f454f64c4e393053f26436ee799c5R28). But that shouldn't stop anyone from testing.

Some changes of note:
* I added a `rebuild` and `rebuild-dev` npm commands because a normal build doesn't actually override the previous files in the `dist` folder because Parcel has cache files. This essentially does a force clear before building. I found these handy to have while developing.
* The sidebar icons in the Editor are now SVG instead of an icon font, they look slightly different and scale differently. I've set a max size to come close to the previous styling.
* The colors are a bit darker/brighter but more uniform. The theming of the newsletter and page presets works a bit differently now (no loose css files). Overall, the styling is now more uniform across the various Editors: mjml newsletter, html newsletter and Page. I have overwritten the hardcoded styles a bit to get the green hover color back, instead of red. 
* The Page Editor sidebar layout changed as some blocks got rearranged and regrouped, so I moved some things around to look more like the previous layout. Works better for existing users, I think.



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. For a new Mautic install (or when using Gitpod), https://github.com/mautic/mautic/issues/13510 (recent merges of 5.x seems to have broken this)
3. Make or edit a new Email: check one written in MJML and one written in html (eg. Paprika and Blank themes)
4. Make or edit a new Landing Page
5. You should still be able to drag&drop Sections and Blocks, the Styles should still work as expected and you can still use the ckeditor to change text
6. Styles don't get lost when opening/closing the Editor
7. Mails/Pages can still be saved and keep their changes when refreshing the page and reopening
